source/common/quic/envoy_quic_server_session.cc

```diff
@@ -191,5 +191,16 @@ quic::QuicSSLConfig EnvoyQuicServerSession::GetSSLConfig() const {
   return config;
 }
 
+void EnvoyQuicServerSession::ProcessUdpPacket(const quic::QuicSocketAddress& self_address,
```

Comment:
Out of curiosity, would we want the Client session to have the same logic that the server session adds in this PR?

Reply:
We are not doing this in any existing QUICHE client. Maybe client CPU usage hasn't been a problem.